### DISCUSSION TITLE
Wiki-links to headings

### DISCUSSION THREAD

**Discussion by Radu-Carlan** [notaapp/talk#411] [Feature Requests] [answered]
From what I can see, the support in Nota for Wiki-links is limited to links to other notes/documents. 

***Any chance this can be extended to links to headings in markdown documents?***

Ideally, I would like to see in Nota something similar to what is available in Obsidian:

- `[[note#heading]]`  --  for links to headings in external documents
- `[[#heading]]`  --  for links to headings in the current document

On a separate (but related) note, I noticed that Nota automatically creates bookmarks (or anchors) for headings by converting all characters to lowercase and replacing spaces with dashes (i.e. `-`). This is great, but it would be even better if it could be configurable to also allow for spaces (and other special characters) used in headings to be simply discarded rather than replaced with dashes. This option would align with the default behaviour in Marked 2. For example, the heading "My Heading" is "my-heading" in Nota, whereas in Marked 2 it is "myheading".

**Comment by astoilkov** [comment, marked as answer]
> Any chance this can be extended to links to headings in markdown documents?

Yes. We should do this. We actually have `[[#heading]]` this syntax in normal markdown links `[]()` we should just add it to wiki-links which should be easy. The linking to notes is easy to make but hard to update. Does the updating of the name of the heading when it changes works in Obsidian? Is it something you need?

> On a separate (but related) note, I noticed that Nota automatically creates bookmarks (or anchors) for headings by converting all characters to lowercase and replacing spaces with dashes (i.e. -). This is great, but it would be even better if it could be configurable to also allow for spaces (and other special characters) used in headings to be simply discarded rather than replaced with dashes. This option would align with the default behaviour in Marked 2. For example, the heading "My Heading" is "my-heading" in Nota, whereas in Marked 2 it is "myheading".

We are following the GitHub convention but I will research this and we can explore that option as well. I think Marked 2 is very popular and it might be a good idea to follow them.

**Reply by Radu-Carlan** [reply]
Thank you for the quick reply, and for taking into account my suggestions.

Obsidian does have the ability to automatically update wiki-links when renaming notes. While this functionality is very convenient, I don't see it as a critical feature. 

As for the GitHub convention for headings, I wasn't suggesting abandoning it in favour of the Marked 2 approach, but rather offering the latter as an option. In fact, you could probably support both (i.e. create both types of anchors for headings, and implement navigation from both).

**Comment by DutchPete** [comment]
> Obsidian does have the ability to automatically update wiki-links when renaming notes.

Luckily Nota has this feature already, and it is essential.

**Comment by DutchPete** [comment]
> I noticed that Nota automatically creates bookmarks (or anchors) for headings by converting all characters to lowercase and replacing spaces with dashes (i.e. `-`).

I am new to Nota and cannot seem to get 'behind' the headings. Can you please tell me how you get to see the bookmarks/anchors?

**Reply by astoilkov** [reply]
You can use a regular markdown link and type `#`. Example: `[](#)`.

**Reply by DutchPete** [reply]
Oh, that's very neat, thank you.

**Reply by DutchPete** [reply]
It only works within the same note/file, not between notes, right?

**Reply by astoilkov** [reply]
Yes.
